viewing_party/party.py

```diff
@@ -1,23 +1,169 @@
+            return  None
 # -----------------------------------------
 # ------------- WAVE 3 --------------------
 # -----------------------------------------
 
-        
+def create_friends_dictionary(user_data):
+    friends_movies = {}
+    friends_movies[1] = user_data["friends"][0]
+    friends_movies[2] = user_data["friends"][1]
+    return friends_movies
+
+def get_unique_watched(user_data):
+    friends_movies = create_friends_dictionary(user_data)
+    users_unique_movies = []
+
+    for title in user_data["watched"]:
+        if title not in friends_movies[1]["watched"] and title not in friends_movies[2]["watched"]:
+            users_unique_movies.append(title)
+    return users_unique_movies
+
+def get_friends_unique_watched(user_data):
+    friends_unique_movies = []
+    friends_movies = create_friends_dictionary(user_data)
+    
+    for movie in friends_movies[1]["watched"]:
+        if movie not in user_data["watched"]:
+            if movie not in friends_unique_movies:
+                friends_unique_movies.append(movie)
+    for friend_movies in friends_movies[2]["watched"]:
+        if friend_movies not in user_data["watched"]:
+            if friend_movies not in friends_unique_movies:
+                friends_unique_movies.append(friend_movies)
+    return friends_unique_movies
+
 # -----------------------------------------
 # ------------- WAVE 4 --------------------
 # -----------------------------------------
+def get_available_recs(user_data):
+    recommendations = []
+    friends_movies = create_friends_dictionary(user_data)
+
+    for movie in friends_movies[1]["watched"]:
+        if movie not in user_data["watched"] and movie["host"] in user_data["subscriptions"]:
+            recommendations.append(movie)
+        for movie in friends_movies[2]["watched"]:
+            if movie not in user_data["watched"] and movie["host"] in user_data["subscriptions"]:
+                recommendations.append(movie)
+    return recommendations
 
 # -----------------------------------------
 # ------------- WAVE 5 --------------------
 # -----------------------------------------
 
+def get_new_rec_by_genre(user_data):
+    recommendations = []
+    friends_movies = create_friends_dictionary(user_data)
+
+    if user_data["watched"] != []:
+        if friends_movies[1]["watched"] != []:
+            for movie in friends_movies[1]["watched"]:
+                if movie not in user_data["watched"] and movie["host"] in user_data["subscriptions"]:
+                    recommendations.append(movie)
+            if friends_movies[2]["watched"] != []:
+                for movie in friends_movies[2]["watched"]:
+                    if movie not in user_data["watched"] and movie["host"] in user_data["subscriptions"]:
+                        recommendations.append(movie)
+            for rec in recommendations:
+                if rec["genre"] not in user_data["watched"][0]["genre"]:
+                    recommendations.remove(rec)

```

Comment:
Notice this falls into the danger zone of modifying a list while iterating over it. If the list happened to have 2 movies back to back with genres not matching the target genre, removing the first would shift the second forward into its old position, then looping would advance the position past the second bad genre movie. So rather than booting out bad values from this list, we might instead copy the _good_ values into a new list to return, or build up the result list only from good values to begin with.

Also, we want to match movies to the user's most popular genre, but this code is comparing to the genre of whatever movie comes first in the list. What if it didn't happen to match the user's most popular genre?